### PR TITLE
SONAR-18408 - Add CirrusCI config for build

### PR DESCRIPTION
**Pull request by benjamin-campomenosi-sonarsource**
## Squashed 
* c596de5 SONAR-18408 - Add CirrusCI config for build

## Changelog 
* d414694 SONAR-18408 - Add CirrusCI config for build
* d82887f remove jetbrains dependency
* c8b3a05 fix build.gradle

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -16,10 +20,20 @@ dependencies {
     compileOnly 'com.google.code.findbugs:jsr305:3.0.2'
 
     implementation 'org.eclipse.jgit:org.eclipse.jgit:6.4.0.202211300538-r'
+    implementation 'org.jetbrains:annotations:13.0'
```
Comment (leo-geoffroy-sonarsource) [inline build.gradle:23]: can we remove this dependency and use Nonnull instead ? (as Duarte PR)

File: build.gradle [hunk 2]
```diff
@@ -1,8 +1,12 @@
 plugins {
     // Apply the java-library plugin for API and implementation separation.
     id 'java-library'
+    id 'com.jfrog.artifactory' version '4.28.1'
+    id "org.sonarqube" version "3.3"
 }
 
+def projectName = 'Git Files Blame'
```
Comment (leo-geoffroy-sonarsource) [inline build.gradle:8]: is it still needed as it is specified in gradle.properties?

File: build.gradle [hunk 3]
```diff
@@ -16,10 +20,20 @@ dependencies {
...
 
     testImplementation 'junit:junit:4.13.2'
+    testImplementation 'org.apache.commons:commons-lang3:3.12.0'
     testImplementation 'org.assertj:assertj-core:3.24.2'
+    testImplementation 'org.mockito:mockito-core:5.1.1'
+}
+
+sonarqube {
+    properties {
+        property 'sonar.projectName', projectTitle
```
Comment (leo-geoffroy-sonarsource) [inline build.gradle:36]: could use projectTitle from gradle.properties?

File: build.gradle (hunk 1 above)
Comment (benjamin-campomenosi-sonarsource) [reply build.gradle]: fix in https://github.com/SonarSource/git-files-blame/commit/d82887f2616f10ac140e472c707ef5eb35efca69

File: build.gradle (hunk 2 above)
Comment (benjamin-campomenosi-sonarsource) [reply build.gradle]: removed in https://github.com/SonarSource/git-files-blame/commit/c8b3a0585d56583c500ca4f3f411fc3a481a13d1

File: build.gradle (hunk 3 above)
Comment (benjamin-campomenosi-sonarsource) [reply build.gradle:33]: looks working good after https://github.com/SonarSource/git-files-blame/commit/c8b3a0585d56583c500ca4f3f411fc3a481a13d1. 👍

**Pull request merged**